src/QuickDrawEngine.cc

```diff
@@ -1022,7 +1023,7 @@ void QuickDrawEngine::pict_write_quicktime_data(StringReader& r, uint16_t opcode
       throw pict_contains_undecodable_quicktime("tiff", move(encoded_data));
     } else {
       throw runtime_error(string_printf(
-          "compressed QuickTime data uses codec %08" PRIX32, desc.codec.load()));
+          "compressed QuickTime data uses codec '%s' [0x%08" PRIX32 "]", string_for_resource_type(desc.codec.load()).c_str(), desc.codec.load()));

```

Comment:
`string_for_resource_type(desc.codec.load())` has to be assigned to a temporary variable here. Someday I'll rethink how string_printf works and eliminate error-prone cases like this :|

Reply:
Strictly speaking, `throw runtime_error(string_printf(...))` also creates a temporary string which is then passed to the constructor of `runtime_error`, and it works ;-)

As a general solution I propose the function `throw_printf(const char* format, ...)` to throw a printf-like message as a `runtime_error`.